Spec/RealtimeClientConnection.swift

```diff
@@ -3366,6 +3366,65 @@ class RealtimeClientConnection: QuickSpec {
                     expect(authorizeMethodCallCount) == 1
                 }
 
+                // RTN22a
+                it("re-authenticate and resume the connection when the client is forcibly disconnected following a DISCONNECTED message containing an error code in the range 40140 <= code < 40150") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    options.useTokenAuth = true
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+                    client.setTransportClass(TestProxyTransport.self)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.Connected) { stateChange in
+                            expect(stateChange?.reason).to(beNil())
+                            done()
+                        }
+                        client.connect()
+                    }
+
+                    guard let initialToken = client.auth.tokenDetails?.token else {
+                        fail("Initial token is nil"); return
+                    }
+
+                    guard let transport = client.transport as? TestProxyTransport else {
+                        fail("TestProxyTransport is not set"); return
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.Disconnected) { stateChange in
+                            guard let reason = stateChange?.reason else {
+                                fail("Error is nil"); done(); return
+                            }
+                            expect(reason.code) == 40140
+                            done()
+                        }
+                    }
+
+                    var authorizeMethodCallCount = 0
+                    let hook = client.auth.testSuite_injectIntoMethodAfter(#selector(client.auth.authorize)) {
+                        authorizeMethodCallCount += 1
+                    }
+                    defer { hook.remove() }
+
+                    let disconnectedMessage = ARTProtocolMessage()
+                    disconnectedMessage.action = .Disconnected
+                    disconnectedMessage.error = ARTErrorInfo.createWithCode(40140, message: "token error (unspecified)")
+                    transport.receive(disconnectedMessage)
```

Comment:
Again, why stub, why not simply issue  a token with a short TTL of say 10 seconds, wait for the disconnect, and then follow the rest of the steps?  I really think we should be using stubbing more sparingly generally, unless it's unavoidable.  In most cases I do not believe it is.